docs/spec/supervisor/supervisor.md

```diff
@@ -0,0 +1,153 @@
+
+```rust
+/// Stores the Light Blocks which could not be confirmed to have originated from
+/// the same chain as the predecesor light blocks.
+type FetchedStack = Stack<LightBlock>;
+
+/// Pair of light block and height of the light block which was highest 
+/// highest one for which we believe originated from the same chain
+type ChainPair = (LightBlock, Height);
```

Comment:
The naming is confusing as it indicates that a `Chain` is paired with something. What I'm also wondering doesn't the `LightBlock` contain its `Height`, what's the benefit of pairing it with the `Height`.

Reply:
Height in this case is the height of the block which was used to check the light. For example if we inserted a light block `A` into the `Trace` after passing all the checks in verify_to_target using light block B of height `h`. The `ChainPair` for `A` would be (`A`, `h`).

Reply:
Is it always the height of `B` -1? In any case the term `Height` is so ubiquitous everywhere that we should encode what height we are referring to here, which in turn an help challenge how it can be constructed.

Reply:
Still the term `ChainPair` for a pairing of block with height is miss-leading, imho.

Reply:
Made a custom `enum VerificationChain`.